src/server/main.js

```diff
@@ -34,6 +37,32 @@ io.on('connect', (socket) => {
       });
   });
 
+  /*
+    registration requires: 'username', 'email', 'password'
+    returns JWT
+   */
+  socket.on('register', async (user) => {
+    const registration = await authentication.register(user);
+    let loggedIn;
+
+    // after registration, login
+    if (registration) {
```

Comment:
If a string is returned by authentication.register (ie. 'username or email already taken' or 'failed to create user'), this if statement still executes. We should only execute this if register returns the boolean true.

Also, if we do get an error on register, we should return that to the client, rather than only emit the result of login.

Reply:
Good catch !
I'll fix that up